native-image-tests/src/main/kotlin/okhttp3/test/graal/JunitTestRegisterer.kt

```diff
@@ -0,0 +1,50 @@
+package okhttp3.test.graal
+
+import com.oracle.svm.core.annotate.AutomaticFeature
+import org.graalvm.nativeimage.hosted.Feature
+import org.junit.platform.engine.discovery.DiscoverySelectors.selectPackage
+import org.junit.platform.engine.support.descriptor.ClassSource
+import org.junit.platform.engine.support.descriptor.MethodSource
+import org.junit.platform.launcher.Launcher
+import org.junit.platform.launcher.TestPlan
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder
+import org.junit.platform.launcher.core.LauncherFactory
+
+@AutomaticFeature
+class JunitTestRegisterer: Feature {
+  override fun beforeAnalysis(access: Feature.BeforeAnalysisAccess?) {
+//    discoverTests()
+  }
+
+  fun discoverTests() {
+    val request = LauncherDiscoveryRequestBuilder.request()
```

Comment:
Idea 2 - detect tests cleanly using junit APIs and register those using an automatic feature.